### PR TITLE
[Merged by Bors] - refactor: golf MvPolynomial.support_finSuccEquiv_nonempty

### PR DESCRIPTION
**Pull request by BoltonBailey**
Refactors the proof of this lemma. Adds a simp lemma to do this, and also adds MvPolynomial analogue

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/MvPolynomial/Equiv.lean [hunk 1]
```diff
@@ -479,16 +479,11 @@ theorem finSuccEquiv_support' {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} :
...
-  simp only [Finset.nonempty_iff_ne_empty, Ne, Polynomial.support_eq_empty]
-  refine fun c => h ?_
-  let ii := (finSuccEquiv R n).symm
-  calc
-    f = ii (finSuccEquiv R n f) := by
-      simpa only [← AlgEquiv.invFun_eq_symm] using ((finSuccEquiv R n).left_inv f).symm
-    _ = ii 0 := by rw [c]
-    _ = 0 := by simp
+  rw [Polynomial.support_nonempty_iff, AddEquivClass.map_ne_zero_iff]
+  exact h
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/MvPolynomial/Equiv.lean:486]: If you're golfing:
```suggestion
  rwa [Polynomial.support_nonempty_iff, AddEquivClass.map_ne_zero_iff]
```

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm either way

File: Mathlib/Data/MvPolynomial/Equiv.lean (hunk 1 above)
Comment (BoltonBailey) [reply Mathlib/Data/MvPolynomial/Equiv.lean]: Nice golf, I've fixed it.

**Comment by BoltonBailey** [comment]
#11655 got merged before this, so I have changed the other lemma name to match that one.

**Comment by erdOne** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**